JSONResourceFile.js

```diff
@@ -203,25 +200,17 @@ JSONResourceFile.prototype.getContent = function() {
  * @private
  */
 JSONResourceFile.prototype._calcLocalePath = function(locale) {
-    var path = "";
-    var language = locale.language;
-    var script = locale.script;
-    var region = locale.region;
-
+    var fullPath = "";
+    var splitLocale = this.locale.getSpec().split("-");
 
-    if (language) {
-        path += language + "/";
-        if (this.baseLocale) {
-            return path;
+    if (this.baseLocale) {
+        fullPath = "/" + splitLocale[0];
+    } else {
+        for (var i=0; i < splitLocale.length; i++) {
+            fullPath += "/" + splitLocale[i];
```

Comment:
How about:

```
fullPath += "/" + splitLocale.join("/");
```

Reply:
Yeah, That's a simpler way. I updated it. Thank you!